pkg/sql/information_schema.go

```diff
@@ -1769,12 +1769,23 @@ func forEachColumnInIndex(
 }
 
 func forEachRole(
-	ctx context.Context, p *planner, fn func(username string, isRole bool) error,
+	ctx context.Context, p *planner, fn func(username string, isRole bool, noLogin bool) error,
 ) error {
-	query := `SELECT username, "isRole" FROM system.users`
+	query := `
+SELECT username, 
+       "isRole",
+       EXISTS 
+       (
+              SELECT option 
+              FROM   system.role_options r 
+              WHERE  r.username = u.username 
+              AND    option ='NOLOGIN') AS noLogin 
```

Comment:
I wasn't following the original role work -- is there a reason why we have to have this all negated? nologin, then negating the result later?

Reply:
One reason is this is how postgres handles it, it stores NOLOGIN rather than LOGIN.
Another reason is it made migration not having to add LOGIN to existing users.